src/classes/OPP_OpportunityAmountSync_SVC.cls

```diff
@@ -0,0 +1,32 @@
+public class OPP_OpportunityAmountSync_SVC {
+
+    /**
+     * Put the fields needed in the query in the description.
+     * @param opportunitiesByIds
+     * @return
+     */
+    public List<Opportunity> syncOppAmountWithTotalPaidPaymentAmount(Map<Id, Opportunity> opportunitiesByIds) {
+        List<Opportunity> oppsToUpdate = new List<Opportunity>();
+        
+        for (Opportunity opp : opportunitiesByIds.values()) {
+            Decimal totalPaid = 0;
+
+            for (npe01__OppPayment__c pmt : opp.npe01__OppPayment__r) {
+                if (pmt.npe01__Paid__c) {
+                    totalPaid += pmt.npe01__Payment_Amount__c == null ? 0 : pmt.npe01__Payment_Amount__c;
+                }
+            }
+
+            Boolean addOppToList = (OPP_StageMappingUtil.determineOppStateFromOppStage(opp.StageName) == 'FINALIZED' && opp.Amount != totalPaid) ||
+                    (OPP_StageMappingUtil.determineOppStateFromOppStage(opp.StageName) != 'FINALIZED' && opp.Amount < totalPaid);
```

Comment:
If we do not move this into the Util, we should reference the constant in the Util class for Finalized.  You might have to change the visibility of the constants, if you do this...

Reply:
Will require updates if we decide to allow negative Opportunity Amounts.  We DO allow negative Payments.